lib/publiccloud/instance.pm

```diff
@@ -286,4 +286,16 @@ sub start
     return $self->wait_for_ssh(timeout => $args{timeout});
 }
 
+=head2 get_state
+
+    get_state();
+
+Get the status of the instance using the CSP api calls.
+=cut
+sub get_state
+{
+    my $self = shift;
+    return $self->provider->get_state_from_instance($self, @_);
```

Comment:
I think it is worth to also add a dummy in `publiccloud::provider` like https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L621 , so if someone calls it on `azure` or `gce` he get a good error message.

Reply:
Oh wow, that `get_state_from_instance()` is implemented in `azure` and `gce` .... so the above comment is only "nice to have" :)

Reply:
Done. But the function exists on `azure` and `gce`, so it should work.